### DISCUSSION TITLE
How to apply anomaly detection on a single folder of Images for segmentation with no labels for supporting labelling

### DISCUSSION THREAD

**Discussion by hrudaykolla** [open-edge-platform/anomalib#2318] [Q&A] [answered]
### What is the motivation for this task?

1. It is a very big task to anomalous images when there are no labels of classification and or segementation when the anomalies are very small.
2. For a model like padim which works on the distribution of the patches, a few anomalous images wont affect the distribution. So applying the model on the folder of images can result with segmentation maps with no metrics.
3. This masks can be used for accurate labelling further.

### Describe the solution you'd like

I would like a data module that accepts the images as a single folder. The results should be segmentation masks of the same images with no metrics as such.

**Comment by abc-125** [comment, marked as answer]
This link in anomalib docs might help you: [how to train with normal images only](https://anomalib.readthedocs.io/en/latest/markdown/guides/how_to/data/custom_data.html#with-only-normal-images), you will have a single folder and synthetic anomalies for validation (or you can skip validation and test step entirely). 
You might also want to take a look [at this paper](https://openaccess.thecvf.com/content/ICCV2023/papers/McIntosh_Inter-Realization_Channels_Unsupervised_Anomaly_Detection_Beyond_One-Class_Classification_ICCV_2023_paper.pdf), Table 3, to see how Padim works with just 10 anomalous images in the training data.
